samples/subsys/fs/src/main.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2018 tavishnaruka@gmail.com
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Sample which uses the filesystem API and SDHC driver */
+
+#include <zephyr.h>
+#include <device.h>
+#include <disk_access.h>
+#include <logging/log.h>
+#include <fs.h>
+#include <ff.h>
+
+LOG_MODULE_REGISTER(main);
+
+static FATFS fat_fs;
+/* mounting info */
+static struct fs_mount_t mp = {
+	.type = FS_FATFS,
+	.fs_data = &fat_fs,
+};
+
+/*
+*  Note the fatfs library is able to mount only strings inside _VOLUME_STRS
+*  in ffconf.h
+*/
+static const char *disk_mount_pt = "/SD:";
+
+void main(void)
+{
+	/* raw disk i/o */
+	do {
+		static const char *disk_pdrv = "SD";
+		u64_t memory_size_mb;
+		u32_t block_count;
+		u32_t block_size;
+
+		if (disk_access_init(disk_pdrv) != 0) {
+			LOG_ERR("Storage init ERROR!");
+			break;
+		}
+
+		if (disk_access_ioctl(disk_pdrv,
+				DISK_IOCTL_GET_SECTOR_COUNT, &block_count)) {
+			LOG_ERR("Unable to get sector count");
+			break;
+		}
+		LOG_INF("Block count %u", block_count);
+
+		if (disk_access_ioctl(disk_pdrv,
+				DISK_IOCTL_GET_SECTOR_SIZE, &block_size)) {
+			LOG_ERR("Unable to get sector size");
+			break;
+		}
+		LOG_INF("Sector size %u", block_size);
+
+		memory_size_mb = block_count * block_size;
+		LOG_INF("Memory Size(MB) %u", (u32_t)memory_size_mb>>20);
+	} while (0);
+
+	mp.mnt_point = disk_mount_pt;
+
+	int res = fs_mount(&mp);
+
+	if (res == FR_OK) {
+		LOG_INF("Disk mounted.");
+	} else {
+		LOG_ERR("Error mounting disk.");
+	}
+
+	while (1) {
+		k_sleep(1000);
```

Comment:
So after the disk is mounted, this sample does nothing but sleep in a loop?  Maybe it could create a file, write something to it, and read it back, or maybe you could augment the instructions to have the user format the disk and add some text files to it, and this sample could get a directory listing of the disk and write out the contents of any .txt files it finds.

Reply:
Added code to list files and directories for now.